crates/sui-core/src/authority.rs

```diff
@@ -1927,14 +1927,13 @@ impl AuthorityState {
     }
 
     pub fn reconfigure(&self, new_committee: Committee) -> SuiResult {
-        // TODO: We should move the committee into epoch db store, so that the operation below
-        // can become atomic.
         fp_ensure!(
             self.epoch() + 1 == new_committee.epoch,
-            SuiError::from("Invalid new epoch to sign and update")
+            SuiError::from("Invalid new epoch")
         );
 
         self.committee_store.insert_new_committee(&new_committee)?;
+        self.db().perpetual_tables.set_epoch(new_committee.epoch)?;
```

Comment:
If this write happens along with `reopen_epoch_db`, why not use `epoch_store.committee.epoch`?